.github/workflows/macos-unit-tests.yml

```diff
@@ -0,0 +1,99 @@
+name: macos-unit-tests
+
+on:
+  workflow_dispatch:
+  pull_request:
+  push:
+    branches:
+      - main
+      - 8.*
+      - 9.*
+
+permissions:
+  contents: read
+
+## Concurrency only allowed in the main branch.
+## So old builds running for old commits within the same Pull Request are cancelled
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: ${{ github.ref != 'refs/heads/main' }}
+
+jobs:
+  unit-tests-push:
+    name: Unit tests ${{ matrix.os }}
+    if: github.event_name == 'push' || github.event_name == 'workflow_dispatch'
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ macos-14 ]
+    runs-on: ${{ matrix.os }}
+    timeout-minutes: 60
+    steps:
+      - uses: actions/checkout@v5
+        with:
+          fetch-depth: 0
+
+      - uses: actions/setup-go@v6
+        with:
+          go-version-file: .go-version
+
+      - name: Install make
+        run: |
+          brew install make
+          export PATH="/usr/local/opt/make/libexec/gnubin:$PATH"
+        shell: bash
+
+      - name: Install mage
+        run: make mage
+        shell: bash
+
+      - name: Beat version
+        run: |
+          beatVersion=$(make get-version)
+          echo "BEAT_VERSION=$beatVersion" >> $GITHUB_ENV
```

Comment:
Couple of things:

1) Why is this needed? As far as I see it no env variable it will default to something, see https://github.com/elastic/elastic-agent/blob/0e533344c831c769d4ecf845978d7074d6149d43/magefile.go#L543-L547

If it's needed, I wonder if this could be merged in the next step for clarity?